src/NuGet.Core/NuGet.Protocol/README.md

```diff
@@ -1 +1,82 @@
-NuGet's implementation for interacting with feeds. Contains functionality for all feed types.
+# NuGet.Protocol
+
+NuGet.Protocol is a NuGet client SDK library that provides a set of APIs for interacting with NuGet feeds. It provides a way for developers to query NuGet feeds to discover packages and their dependencies, and also to download packages and their associated assets.
+
+## Usage
+
+At the center of this library are the PackageSource and SourceRepository types, which represent a NuGet source that may be a file source or an http based source implementing the V2 or [V3](https://learn.microsoft.com/nuget/api/overview#versioning) protocol.
+
+```
+PackageSource localSource = new PackageSource(@"D:\LocalSource");
+SourceRepository localRepository = Repository.Factory.GetCoreV3(localSource);
+
+SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+```
+
+The SourceRepository then has a GetResourceAsync method that you can use to acquire implementations of INuGetResource that often are [V3](https://learn.microsoft.com/nuget/api/overview#versioning) resources.
```

Comment:
NuGet.Protocol's resource names do not match the server API resource names (the APIs are also not a 1:1 mapping). So, I think it would be useful to list the "base" resource types in the readme, to make it much easier for customers to figure out which resource types are likely to be useful.

Reply:
mapping is a great idea.

Here's some pointers if you decide to add it.

FindPackagesById = content/flat container
PackageSearchResource = search
PackageMetadataResource = package metadata resources/registration
PackageUpdateResource = push and dlete

Reply:
+1 for this.